### PR TITLE
Refactor attributes

### PR DESCRIPTION
**Pull request by beauby**

### CODE REVIEW

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -159,15 +157,24 @@ def json_key
       root || object.class.model_name.to_s.underscore
     end
 
+    def attribute_value(name)
+      if self.class._attributes_values[name]
+        instance_eval(&self.class._attributes_values[name])
+      elsif respond_to?(name) # To handle legacy method-based attr overriding
+        warn 'Overriding attributes by defining a method on the serializer is deprecated. Please use the block syntax.'
```
Comment (NullVoxPopuli) [inline lib/active_model/serializer.rb]: can this have a reference to more information about this?

File: test/serializers/attribute_test.rb [hunk 2]
```diff
@@ -71,6 +71,38 @@ def id
...
+      PostWithSelectAndTest = Class.new(::Model)
+      class PostWithSelectAndTestSerializer < ActiveModel::Serializer
+        attribute :select do
+          'select'
+        end
+        attribute :test do
+          'test'
+        end
+      end
+      def test_reserved_works_attribute
```
Comment (NullVoxPopuli) [inline test/serializers/attribute_test.rb]: does there need to be a newline between the end/def here?

**Comment by NullVoxPopuli** [comment]
needs rebase

**Comment by beauby** [comment]
@NullVoxPopuli Will rebase and improve when/if we merge #1262 (which this PR relies on).

**Comment by bf4** [comment]
@beauby Is there anything in this PR that remains to be done against master?

**Comment by beauby** [comment]
Closing as equivalent changes have been merged from other PRs.

**Pull request closed without merging**